src/sentry/digests/utils.py

```diff
@@ -63,20 +64,31 @@ def get_digest_as_context(digest: Digest) -> Mapping[str, Any]:
 
 
 def get_events_by_participant(
-    participants_by_provider_by_event: Mapping[Event, Mapping[ExternalProviders, set[Team | User]]]
-) -> Mapping[Team | User, set[Event]]:
+    participants_by_provider_by_event: Mapping[
+        Event, Mapping[ExternalProviders, set[Team | APIUser]]
+    ]
+) -> Mapping[Team | APIUser, set[Event]]:
     """Invert a mapping of events to participants to a mapping of participants to events."""
     output = defaultdict(set)
+    api_user_cache: Dict[int, APIUser] = {}
```

Comment:
@corps this is the only weird change I had to make to get all these changes working. Some tests are still getting users into  `participants_by_provider_by_event`